packages/flutter/test/gestures/gesture_binding_resample_event_on_widget_test.dart

```diff
@@ -0,0 +1,118 @@
+  // blocked by [AutomatedTestWidgetsFlutterBinding]. Override this behavior
+  // to help events go into widget tree.
+  @override
+  void dispatchEvent(
+    PointerEvent event,
+    HitTestResult hitTestResult, {
+    TestBindingEventSource source = TestBindingEventSource.device,
+  }) {
+    super.dispatchEvent(event, hitTestResult, source: TestBindingEventSource.test);
+  }
+}
+
+void main() {
+  final TestWidgetsFlutterBinding binding = PointerDataAutomatedTestWidgetsFlutterBinding();
+  testWidgets('PointerEvent resampling on a widget', (WidgetTester tester) async {
+    assert(WidgetsBinding.instance == binding);
+    Duration now() => Duration(milliseconds: binding.clock.now().millisecondsSinceEpoch);
+    final Duration epoch = now();
+    final ui.PointerDataPacket packet = ui.PointerDataPacket(
+      data: <ui.PointerData>[
+        ui.PointerData(
+            change: ui.PointerChange.add,
+            physicalX: 0.0,
+            timeStamp: epoch + const Duration(milliseconds: 0),
+        ),
+        ui.PointerData(
+            change: ui.PointerChange.down,
+            physicalX: 0.0,
+            timeStamp: epoch + const Duration(milliseconds: 1),
+        ),
+        ui.PointerData(
+            change: ui.PointerChange.move,
+            physicalX: 10.0,
+            timeStamp: epoch + const Duration(milliseconds: 2),
+        ),
+        ui.PointerData(
+            change: ui.PointerChange.move,
+            physicalX: 20.0,
+            timeStamp: epoch + const Duration(milliseconds: 3),
+        ),
+        ui.PointerData(
+            change: ui.PointerChange.move,
+            physicalX: 30.0,
+            timeStamp: epoch + const Duration(milliseconds: 4),
+        ),
+        ui.PointerData(
+            change: ui.PointerChange.up,
+            physicalX: 40.0,
+            timeStamp: epoch + const Duration(milliseconds: 5),
+        ),
+        ui.PointerData(
+            change: ui.PointerChange.remove,
+            physicalX: 40.0,
+            timeStamp: epoch + const Duration(milliseconds: 6),
+        ),
+      ],
+    );
+
+    final List<PointerEvent> events = <PointerEvent>[];
+    await tester.pumpWidget(
+      Directionality(
+        textDirection: TextDirection.ltr,
+        child: Listener(
+          onPointerDown: (PointerDownEvent event) => events.add(event),
+          onPointerMove: (PointerMoveEvent event) => events.add(event),
+          onPointerUp: (PointerUpEvent event) => events.add(event),
+          child: const Text('test'),
+        ),
+      ),
+    );
+
+    GestureBinding.instance.resamplingEnabled = true;
+    GestureBinding.instance.samplingOffset = const Duration(microseconds: -5500);
+    ui.window.onPointerDataPacket(packet);
+    expect(events.length, 0);
+
+    await tester.pump(const Duration(milliseconds: 7));
+    expect(events.length, 1);
+    expect(events[0].runtimeType, equals(PointerDownEvent));
+    expect(events[0].timeStamp, now() + binding.samplingOffset);
```

Comment:
Nit: I guess `binding.samplingOffset` here should be `const Duration(microseconds: -5500)`? I might be clearer to create a constant and use it in line 93, 100, 106, and 114.

Reply:
Using `binding.samplingOffset` is in David's version. I changed it to `kSamplingOffset`, but I don't get why this is better.

Reply:
I think it's a little better because
1. if the binding somehow gets the `samplingOffset` wrong, the `kSamplingOffset` version of `expect` would catch that while the `binding.sampleOffset` version of `expect` won't.
2. David's original version uses `GestureBinding.instance.samplingOffset` which would match the `GestureBinding.instance.samplingOffset = kSamplingOffset` in a simple text search. In the new version, `binding` isn't immediately obvious to be the same as `GestureBinding.instance` (but I believe they're the same).